posts/mobs-are-not-always-destructive.md

```diff
@@ -0,0 +1,94 @@
+
+So where does mob programming, or "mobbing", come into the picture? Consider the following situation. During a pairing session between two rather opinionated engineers, neither person will concede on their stance for the best course of action. They require outside consultation from their very busy lead. The lead comes in later and listens to both explain their rationales. While they do that, a team member nearby listens in and can not help but interject themselves having just worked previously on the service class in question. What was a two person effort has now bloomed into a four person venture. And of course there is still functionally only one screen and one keyboard/mouse combination.
+
+One person now sits at the computer, "driving", while the other three look over his/her shoulder. They all exchange on how best to move forward and then make sure the individual at the computer understands what the team agreed upon as it is typed out. I is my belief, that even if someone has not participated in paired programming previously, we can all recall times like these in our professional careers. Mob programming is just the name we have given this situation.
+
+# There are... Guidelines
+
+Mobbing is meant to be an inclusive exercise so that no one individual on the team is always dictating what to do. Guidelines were created, therefore, to help enforce that mentality. They are as follows:
+
+1. Everyone focused on the goal (if you have to do something else step out)
+1. One person coding at a time
+1. Person coding switches every N minutes
+1. Problem solve as a group
+1. Person on the keyboard does not type anything they do not understand
+
+The whole group is centered around completing a single goal. Since only one individual is driving at a time at the keyboard, this allows for the entire team to participate, share, and collaborate on the group's solution.
+
+# How a Whole Mob can be Agile
+
+Worth highlighting that mob programming first was mentioned in "[Chapter 28: Mob Programming and the Transition to XP](https://www.worldcat.org/title/extreme-programming-perspectives/oclc/49663563)" of the _Extreme Programming Perspectives_ by Moses Hohman in 2003. However, just because it originated in the Extreme Programming (XP) methodology does not mean it cannot also be applied to other methodologies like scrum, kanban, scrum-ban, lean, etc. and here are two reasons why that is the case:
```

Comment:
```suggestion
It's worth highlighting that mob programming first was mentioned in "[Chapter 28: Mob Programming and the Transition to XP](https://www.worldcat.org/title/extreme-programming-perspectives/oclc/49663563)" of the _Extreme Programming Perspectives_ by Moses Hohman in 2003. However, just because it originated in the Extreme Programming (XP) methodology does not mean it cannot also be applied to other methodologies like scrum, kanban, scrum-ban, lean, etc. and here are two reasons why that is the case:
```

Reply:
I feel like mentioning the agile implementations here feels off

Reply:
OK that's a good note. And what feels off about it?

Reply:
eh feels unnecessary to tell people that they are allowed to apply some coding technique to an agile implementation. Do you though